sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -905,7 +905,9 @@ object SparkSession extends Logging {
       // Get the session from current thread's active session.
       var session = activeThreadSession.get()
       if ((session ne null) && !session.sparkContext.isStopped) {
-        options.foreach { case (k, v) => session.sessionState.conf.setConfString(k, v) }
+        for ((k, v) <- options if !SQLConf.staticConfKeys.contains(k)) {
```

Comment:
Could we have clearer warning message than `configuration may not take effect` if `options` has any static config?

Reply:
+1

Reply:
+1 too. we can warn static SQL separately. But do we still need a general warning messages for core configurations?
e.g.

```scala
for ((k, v) <- options) {
        if (SQLConf.staticConfKeys.contains(k)) {
          logWarning(s"Using an existing SparkSession, the static configuration $k is ignored.")
        } else {
          session.sessionState.conf.setConfString(k, v)
        }
      }
      if ((options -- SQLConf.staticConfKeys.asScala).nonEmpty) {
        logWarning("Using an existing SparkSession; some spark core configurations may not take" +
          " effect.")
      }
```

Reply:
+1 too. we can warn static SQL separately. But do we still need a general warning messages for core configurations?
e.g.

```scala
for ((k, v) <- options) {
        if (SQLConf.staticConfKeys.contains(k)) {
          logWarning(s"Using an existing SparkSession, the static configuration $k is ignored.")
        } else {
          session.sessionState.conf.setConfString(k, v)
        }
      }
      if ((options -- SQLConf.staticConfKeys.asScala).nonEmpty) {
        logWarning("Using an existing SparkSession; some spark core configurations may not take" +
          " effect.")
      }
```